### ISSUE TITLE
Stopping and removing containers breaks docker access through WSL2

### ISSUE THREAD

**Issue by vnijs** [docker/for-win#7836]
Running the below bash code consistently breaks access to Docker from WSL2. Is that expected? If so, what is the best way to get Docker up and running again?

```
running=$(docker ps -q)
if [ "${running}" != "" ]; then
  echo "Stopping running containers ..."
  docker stop ${running}
else
  echo "No running containers"
fi

imgs=$(docker images | awk '/<none>/ { print $3 }')
if [ "${imgs}" != "" ]; then
  echo "Removing unused containers ..."
  docker rmi ${imgs}
else
  echo "No images to remove"
fi

procs=$(docker ps -a -q --no-trunc)
if [ "${procs}" != "" ]; then
  echo "Removing errand docker processes ..."
  docker rm ${procs}
else
  echo "No processes to purge"
fi
```

**Comment by vnijs** [comment]
Just to clarify ... Docker initially runs fine with Ubuntu 20.04. But after running the bash code above, the output from `docker ps` is:

```
docker ps

The command 'docker' could not be found in this WSL 2 distro.
We recommend to activate the WSL integration in Docker Desktop settings.

See https://docs.docker.com/docker-for-windows/wsl/ for details.
```

![image](https://user-images.githubusercontent.com/2998956/88537870-36039c00-cfc3-11ea-86dc-7ebcfc42ad2a.png)

**Comment by stephen-turner** [comment]
No, that doesn't seem normal. Which of the three stanzas in your bash script breaks it?

**Comment by vnijs** [comment]
Minimal reproducible example below. Docker stops working inside WSL2 after running this script. Likely the same issue I reported in https://github.com/docker/for-win/issues/7803

```
#!/bin/bash

docker run -d -v /mnt:/mnt ubuntu

running=$(docker ps -q)
docker stop ${running}

procs=$(docker ps -a -q --no-trunc)
docker rm ${procs}
```

![image](https://user-images.githubusercontent.com/2998956/88583898-b3e89700-d005-11ea-8c91-faf36cebd9b4.png)

![image](https://user-images.githubusercontent.com/2998956/88583691-6c620b00-d005-11ea-943f-2514b39a2bad.png)

![image](https://user-images.githubusercontent.com/2998956/88584204-26f20d80-d006-11ea-9e18-7c0f532a28d1.png)

**Comment by stephen-turner** [comment]
We have another report of this at #7901.

**Comment by stephen-turner** [comment]
We will look into this, but it may not be for a couple of weeks, given the vacation schedules of the relevant engineers.

**Comment by chiptus** [comment]
The command the breaks docker for me is `docker rm container`

**Comment by StefanScherer** [comment]
I have at least on repro

<img width="1441" alt="dd-docker-command-missing-repro-7836" src="https://user-images.githubusercontent.com/207759/89024250-17840480-d325-11ea-9e14-01be5ed737d7.png">

```
$ docker run -d -v /mnt:/mnt ubuntu sleep 60
$ docker ps
$ running=$(docker ps -q)
$ docker stop ${running}
$ procs=$(docker ps -a -q --no-trunc)
$ docker rm ${procs}
$ docker
-bash: /usr/bin/docker: No such file or directory
```

Created and uploaded diagnostics:  0551A6EC-4754-4106-90EA-154CE1A32245/20200731095248

**Comment by stephen-turner** [comment]
Strange, I couldn't reproduce it. I wonder what the difference is.

**Comment by dennispries-awdev** [comment]
I have a similar issue with a compose file that has 25 Services in it (sadly some private repos, so i cannot share it here). The services are updated every night. Always when I do my update with "docker-compose pull" and then "docker-compose up -d" it is going to remove the services and then fails recreating it with the error 

`Cannot create container for service io-module-core: dial unix /mnt/wsl/d`
`Recreating a6b86ab55957_awiot_io-module-s3_1          ... error`
`ERROR: for awiot_s3_1  b'dial unix /mnt/wsl/docker-desktop/shared-sockets/guest-services/docker.sock: connect: no such file or directory'`

After that, the following folders have some empty files in it and are no longer mounted:

/mnt/wsl/docker-desktop
/mnt/wsl/docker-desktop-bind-mounts
/mnt/wsl/docker-desktop-desktop-data

Because WSL does not seem to remount them if they are not empty, I have to remove them and restart Docker Desktop.

If I do a docker-compose down he manages to remove all containers, but failes in removing the network (same error, that the docker.sock cannot be found)

I have to do the procedure above again, do a docker-compose down to remove the network and after a docker-compose up -d everythings fine again.

I cannot see any errors in the docker log file:
[log.txt](https://github.com/docker/for-win/files/5193686/log.txt)

It seams that the pathes above gets unmounted. If it helps, I've run the diagnostics directly after it happens, the ID is:
`117479ED-71EA-4EFE-AA2E-D85E3A9C67F5/20200909082136`

**Comment by djs55** [comment]
This issue should be fixed in [stable 2.4.0.0](https://desktop.docker.com/win/stable/48506/Docker%20Desktop%20Installer.exe).

If you have further issues, please open fresh tickets. Thanks for your reports!
